Codecademy/gamut
# feat(badge): adding `leading-icon` to badge

### Overview
Adding an icon prop to the badge component. 
Also adding a tertiary-fill variant for badges. 

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?type=design&node-id=29959-39721&mode=design&t=nBBLirSlYHPh47rh-0
- [x] Related to JIRA ticket: [GM-591](https://codecademy.atlassian.net/browse/GM-591)
- [x] Related to JIRA ticket: [GM-642](https://codecademy.atlassian.net/browse/GM-642)

#### Testing Instructions
For Mono: 
1. For the tertiaryFill and icon, you can go to the [homepage](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-6193)
2. Then check that badges aren't broken, go to [the pricing page](https://tengu.codecademy.com/pricing), there's a "New" badge above "AI learning assistance" -- check that it renders just fine 

For the [monolith](https://pr-37659-monolith.dev-eks.codecademy.com/admin): 
1. Log in with an admin account
2.  Go to the admin page
3. check that both badges ("For Kicks" + "and giggles") render


#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37659)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Mono       | [Mono Link](https://github.com/codecademy-engineering/mono/pull/6193)  | [Portal Env](http://www.google.fr/ 'Named link title')   




[GM-591]: https://codecademy.atlassian.net/browse/GM-591?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[GM-642]: https://codecademy.atlassian.net/browse/GM-642?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
